lib/skate/detours/detours.ex

```diff
@@ -61,29 +65,9 @@ defmodule Skate.Detours.Detours do
   end
 
   @spec db_detour_to_detour(Detour.t()) :: DetailedDetour.t() | nil
-  def db_detour_to_detour(
-        %{
-          state: %{
-            "context" => %{
-              "route" => %{"name" => route_name, "directionNames" => direction_names},
-              "routePattern" => %{"headsign" => headsign, "directionId" => direction_id},
-              "nearestIntersection" => nearest_intersection
-            }
-          }
-        } = db_detour
-      ) do
-    direction = Map.get(direction_names, Integer.to_string(direction_id))
-
-    %DetailedDetour{
-      id: db_detour.id,
-      route: route_name,
-      direction: direction,
-      name: headsign,
-      intersection: nearest_intersection,
-      updated_at: timestamp_to_unix(db_detour.updated_at),
-      author_id: db_detour.author_id,
-      status: categorize_detour(db_detour)
-    }
+  @spec db_detour_to_detour(status :: detour_type(), Detour.t()) :: DetailedDetour.t() | nil
+  def db_detour_to_detour(%{} = db_detour) do
+    db_detour_to_detour(categorize_detour(db_detour), db_detour)
   end
 
   def db_detour_to_detour(invalid_detour) do
```

Comment:
This has kinda been replaced by `db_detour_to_detour/2` on lines 101-7, because the only reason this function would execute is if the detour is simply not a map at all (probably if it's nil?) I'm not sure this function would still be useful in that case, but if it is, then the error message should be changed

Reply:
I'm still thinking about this, but I'm going to re-request review while I consider this, to unblock the other conversations in case I can iterate on those faster.

Reply:
Removed the function for now, I think we'll be coming back though here for the refactor anyway so I think that'll be fine for now https://github.com/mbta/skate/compare/752d652b63447cf46f6a5a6aabcdb0f3eb36a98f..f88875b154cd8218a2eadca2adce531963439e29